src/components/controller/controller.ts

```diff
@@ -0,0 +1,44 @@
+import { ResponceModelAll, ResponceModelSources } from '../models/index.model';
+import AppLoader from './appLoader';
+
+interface Callback {
+  (data: ResponceModelSources | ResponceModelAll): void;
+}
+class AppController extends AppLoader {
+  getSources(callback: Callback) {
+    super.getResp(
+      {
+        endpoint: 'sources',
+      },
+      callback
+    );
+  }
+
+  getNews(e: Event, callback: Callback) {
+    let target: HTMLDivElement | null = e.target instanceof HTMLDivElement ? e.target : null;
```

Comment:
you can omit type declaration `HTMLDivElement | null` as it will be inferred form the right part.